exec/cnex/array.c

```diff
@@ -47,6 +70,30 @@ void array_freeArray(Array *self)
     free(self);
 }
 
+void array_clearArray(Array *self)
+{
+    if (self != NULL && self->data != NULL) {
+        for (size_t i = 0; i < self->size; i++) {
+            cell_clearCell(&self->data[i]);
+        }
+        free(self->data);
+    }
+    self->data = NULL;
+    self->size = 0;
+
+    array_expandArray(self, 0);
+}
+
+void array_removeItem(Array *self, size_t index)
+{
+    if (self != NULL && self->data != NULL) {
+        for(size_t i = index; i < self->size - 1;i++) {
```

Comment:
If this is ever called with self->size == 0, there's going to be a major problem here because size is unsigned.

Reply:
Added sanity checked for size to be > 0.